Update Stack.yaml and Cabal Files

The lastest stack and cabal files are from 6 months ago and were generated from the 1.6.0 build.

The gi-pango 1.0.22 from that build has a bug where it parses harfbuzz incorrectly as reported here
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=968729
and got fixed in gi-pango 1.0.23

The gi-pango bug is also reported here:
https://aur.archlinux.org/packages/deadd-notification-center/

My best guess is that current 1.7.2 was built on a different stack LTS GHC than the latest stack.yaml and .cabal files posted in the repo.

```
gi-pango                 > [85 of 94] Compiling GI.Pango.Objects.Font
gi-pango                 >
gi-pango                 > /tmp/stack-69059c6a2ea7a9dc/gi-pango-1.0.22/GI/Pango/Objects/Font.hs:669:9: error:
gi-pango                 >     parse error on input ‘HarfBuzz.FeatureT.feature_t’
gi-pango                 >     |
gi-pango                 > 669 |     Ptr HarfBuzz.FeatureT.feature_t ->      -- features : TCArray False (-1) 2 (TInterface (Name {namespace = "HarfBuzz", name = "feature_t"}))
gi-pango                 >     |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
gi-pango                 >

```

--- a/stack.yaml
+++ b/stack.yaml
@@ -41,25 +41,27 @@ packages:
 # (e.g., acme-missiles-0.3)
 extra-deps: [ ConfigFile-1.1.4
             , system-locale-0.3.0.0
+            , gi-harfbuzz-0.0.3
             , dbus-0.10.15
             , gtk3-0.15.4
             , stm-2.5.0.0
             , hgettext-0.1.31.0
-            , glib-0.13.8.0
-            , pango-0.13.8.0
+            , glib-0.13.8.1
+            , pango-0.13.8.1
             , gio-0.13.8.0
-            , gi-pango-1.0.22
-            , haskell-gi-0.23.0
-            , haskell-gi-base-0.23.0
-            , gi-glib-2.0.23
-            , gi-cairo-1.0.23
-            , gi-atk-2.0.21
-            , gi-gdk-3.0.22
-            , gi-gdkpixbuf-2.0.23
-            , gi-gio-2.0.26
-            , gi-gobject-2.0.22
-            , gi-gtk-3.0.32
-            , gi-graphene-1.0.1
+            , gi-pango-1.0.23
+            , haskell-gi-0.24.5
+            , haskell-gi-base-0.24.4
+            , gi-glib-2.0.24
+            , gi-cairo-1.0.24
+            , gi-atk-2.0.22
+            , gi-gdk-3.0.23
+            , gi-gdkpixbuf-2.0.24
+            , gi-gio-2.0.27
+            , gi-gobject-2.0.23
+            , gi-gtk-3.0.36
+            , gi-graphene-1.0.2
+            , ansi-terminal-0.11
             ]
 
 # Override default flag values for local packages and extra-deps

--- a/stack.yaml
+++ b/stack.yaml
@@ -58,7 +58,7 @@ extra-deps: [ ConfigFile-1.1.4
             , gi-gdk-3.0.23
             , gi-gdkpixbuf-2.0.24
             , gi-gio-2.0.27
-            , gi-gobject-2.0.23
+            , gi-gobject-2.0.25
             , gi-gtk-3.0.36
             , gi-graphene-1.0.2
             , ansi-terminal-0.11

